### ISSUE TITLE
Support ruby 2.1+ Exception#cause

### ISSUE THREAD

**Issue by eoinkelly** [MindscapeHQ/raygun4ruby#70]
Ruby `Exception` has a `#cause` method since ruby 2.1 that returns the exception that was active at the time the current exception was raised. This makes debugging really nice.

It would be great if Raygun could report the cause chain on versions of ruby that support it.

**Comment by nikz** [comment]
We could add this to `customData` perhaps?

**Comment by eoinkelly** [comment]
Seems reasonable. The nested exceptions will have a stack too which we would probably want to see. This would probably increase payload size and processing time so perhaps it should be opt-in (at least at first)

**Comment by CmdrKeen** [comment]
I'm going to have one of our team look into this. We already handle nested exceptions in other providers (.NET, Java etc) so we should just re-use that known data structure.

**Comment by eoinkelly** [comment]
:sparkles:

**Comment by nikz** [comment]
Example stolen from [Avdi Grimm's Blog](http://devblog.avdi.org/2013/12/25/exception-causes-in-ruby-2-1/)

```ruby
begin
  begin
    raise "Error A"
  rescue => error
    raise "Error B"
  end
rescue => error
  puts "Current failure: #{error.inspect}"
  puts "Original failure:  #{error.cause.inspect}"
end
# >> Current failure: #<RuntimeError: Error B>
# >> Original failure:  #<RuntimeError: Error A>
```

**Comment by fundead** [comment]
Cheers for that, I've got this queued up to be looked at soon, needs backend work to include the additional data in the identity but it's a straight port from other langs so shouldn't be too much required.

**Comment by UberMouse** [comment]
Resolved by #107
